mainloop.cpp

```diff
@@ -390,8 +390,12 @@ bool Mainloop::add_endpoints(Mainloop &mainloop, struct opt *opt)
         g_tcp_fd = tcp_open(opt->tcp_port);
 
     if (opt->logs_dir) {
-        _ulog = new ULog(opt->logs_dir);
-        g_endpoints[i] = _ulog;
+        if (streq(DIALECT, "common")) {
+            _log_endpoint = new ULog(opt->logs_dir);
+        } else {
```

Comment:
Currently, build can only be done with `common` or `ardupilotmega` - so it should be safe - for now, at least.